patientsearch/api.py

```diff
@@ -162,6 +164,28 @@ def resource_by_id(resource_type, resource_id, methods=["GET"]):
     return jsonify(HAPI_request(resource_type, resource_id, token))
 
 
+def resource_from_args(resource_type, args):
+    """Generate FHIR resource from given type and args"""
+    # refactor into model as need expands
+    current_app.logger.debug(f"generate {resource_type} from {args}")
+    if resource_type != 'Patient':
+        raise ValueError("only Patient aware ATM")
+
+    # The birthdate includes HAPI search syntax; parse from a pattern such as:
+    #   Patient.birthdate=eq1977-01-12
+    dob = re.match(r'eq(.*)', args.get('subject:Patient.birthdate'))
```

Comment:
Tiny insignificant nitpick, but for such a simple use-case, I prefer `split()` or string slicing instead of regular expressions

Reply:
As suggested - refresh.